Restore wisp-mode and update url to hg.sr.ht

ArneBab opened the pull request:
Wisp mode was removed in commit 6d7595adb2a4b762fd5a320cc31e253fedee28c1 because the bitbucket repos were killed by bitbucket. wisp-mode moved to hg.sr.ht; this reflects the change.

### Brief summary of what the package does

wisp-mode provides highlighting and editing support for wisp-files. See https://www.draketo.de/english/wisp

### Direct link to the package repository

https://hg.sr.ht/~arnebab/wisp

### Your association with the package

Creator and maintainer

### Relevant communications with the upstream package maintainer

Restores the package.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
As far as I can tell the [old wisp-mode](https://github.com/emacsattic/wisp-mode) only contained one .el file but this seems to include ob-wisp.el, which should probably be in its own package; should the recipe be updated to reflect this?

Also per the latest MELPA guidelines, please add a LICENSE file to the repository.

There are a few very minor issues around the hook (you shouldn't need to `defvar` it since define-derived-mode will define it for you; using a hook to silently turn off electric-indent-mode could surprise users) but those are definitely possible to grandfather in, I think.

ArneBab commented:
Thank you for your review!

The ob-wisp.el file is still work in progress — I’d love to know how to adapt the package to include it once ob-wisp.el actually works..

Is it sufficient to make the LICENSE file a symlink to a COPYING file (I create it via autotools so that’s a standard file)?

What would be a better approach to disable electric-indent-local-mode in wisp-mode?

riscy commented:
> The ob-wisp.el file is still work in progress — I’d love to know how to adapt the package to include it once ob-wisp.el actually works..

What you'd want to do in this case is make a separate package, but create a dependency (using a Package-Requires line) if one is necessary.  The reason for separate packages is that each package should have its own strict namespace to prevent confusion and collisions.  Note that two packages can exist in the same repository, but you would have to modify your MELPA recipe to exclude certain files.

> Is it sufficient to make the LICENSE file a symlink to a COPYING file (I create it via autotools so that’s a standard file)?

Oh yes, you don't even need to symlink it.  I should have specified.  Now I see you already do have a gpl.txt file, but I think it will be more easily detected if it's named COPYING or LICENSE.

> What would be a better approach to disable electric-indent-local-mode in wisp-mode? (though I’m currently not sure whether this is still needed with the more recent changes to wisp-mode; I currently do not see problems when I enable it)

If there are no problems these days, I would just go with the simplest solution and remove the hook.  If you want to inhibit electric-indent-mode, I think this is the correct way:

```
electric-indent-inhibit is a variable defined in ‘electric.el’.
Its value is t
Local in buffer melpazoid.py; global value is nil

  Automatically becomes buffer-local when set.

Documentation:
If non-nil, reindentation is not appropriate for this buffer.
This should be set by major modes such as ‘python-mode’ since
Python does not lend itself to fully automatic indentation.
```

riscy commented:
Ping when ready for another pass. :)

ArneBab commented:
will do — I’m currently stuck on otherr projects.

ArneBab commented:
I just added the fixes to wisp-mode and changed the package to only include wisp-mode.el

- the license is now in COPYING
- use electric-indent-inhibit instead of electric-indent-local-mode
- use the variable defined by define-derived-mode

riscy commented:
> the license is now in COPYING

I looked into your repo and still see it at gpl.txt, but I think that's fine.  Putting it in a file like LICENSE or COPYING is closer to standard and so the license might be easier to machine detect.

Taking another look at the code, you shouldn't need to add a hook to your own mode; unless I'm missing something
```elisp
;;;###autoload
(add-hook 'wisp-mode-hook
          (lambda ()
            (setq electric-indent-inhibit t)))
```
can just go under your `define-derived-mode` form.  But since this is adding a hook to your own mode, I don't think that's a blocker.

Thank you for this, and welcome to MELPA. :)

ArneBab commented:
Thank you!

I just checked locally: I had forgotten to actually `add` the COPYING symlink. Sorry …
